### PR TITLE
Updated to SpongeAPI 6.1

### PR DESCRIPTION
**Pull request by maxanier**
As the title says this PR updates nations to compile with API6.1
Main problem was that many events do not have a getTargetWorld() method since somehow one event can now contain multiple blocks/etc from different worlds????? Not sure why
https://github.com/SpongePowered/SpongeAPI/commit/403a0d8c1d5684c1e6b2f6505f7e3cb1883e9bb0

Hopefully I did not introduce any new bugs.

PR #123 Did not work for me as it only updated a few listeners.

Build/Download: https://github.com/maxanier/Nations/releases/tag/v2.4-API6.1

### CODE REVIEW

File: src/main/java/com/arckenver/nations/listener/BuildPermListener.java [hunk 1]
```diff
@@ -16,17 +16,15 @@
...
 {
 
 	@Listener(order=Order.FIRST)
 	public void onPlayerPlacesBlock(ChangeBlockEvent.Place event, @First Player player)
 	{
-		if (!ConfigHandler.getNode("worlds").getNode(event.getTargetWorld().getName()).getNode("enabled").getBoolean())
-		{
-			return;
-		}
+
```
Comment (TheoKah) [inline src/main/java/com/arckenver/nations/listener/BuildPermListener.java:27]: Why did you remove this part where we check if the plugin is enabled in the world?

If you think doing this elsewhere is better we can discuss it, but in that case, we would need to change it everywhere for consistency sake

File: build.gradle [hunk 2]
```diff
@@ -1,9 +1,10 @@
 apply plugin: 'java'
 apply plugin: 'eclipse'
+apply plugin: 'idea'
```
Comment (TheoKah) [inline build.gradle:3]: please do not commit this file. I'll deal with the version changes / api compilation

**Review by TheoKah — CHANGES_REQUESTED** [review]
Can you change/discuss the couple things I pointed out?

**Comment by TheoKah** [comment]
I made a branch dedicated to api 6, as it is still beta, api 5 will remain on branch master

Maybe we can even jump to 7 as 6 is unsupported by sponge

File: src/main/java/com/arckenver/nations/listener/BuildPermListener.java (hunk 1 above)
Comment (maxanier) [reply src/main/java/com/arckenver/nations/listener/BuildPermListener.java:27]: Since it is possible that one event contains multiple block changes at ones according to https://github.com/SpongePowered/SpongeAPI/commit/403a0d8c1d5684c1e6b2f6505f7e3cb1883e9bb0, which do not have to be in the same world. (No idea why)
Because you are looping through all transactions afterwards, I decided to move the 'enabled world' check to that loop to test it for each transaction (possibly in a different world).

I did apply this behavior to all changed events except for the entity spawn event as you only check the permission for the first entity anyway

**Comment by maxanier** [comment]
If you agree with my comment regarding the requested changes on the BuildPermListener, I will reset the build.gradle and resolve the conflicts.

**Comment by TheoKah** [comment]
Sorry for late answer. Yep, I'm ok with the changes about events

**Comment by maxanier** [comment]
Replaced by #142
